packages/esm-patient-test-results-app/src/filter/filter-reducer.ts

```diff
@@ -19,7 +19,7 @@ const computeParents = (prefix, node) => {
     });
     let activeTests = [];
     node.subSets.forEach((leaf) => {
-      if (leaf.obs.length) {
+      if (leaf.obs?.length) {
```

Comment:
I'd do `Array.isArray(leaf?.obs) && leaf.obs.length > 0` - this way you not only mitigate `leaf` for some reason being undefined, but you also make sure that the object is indeed an `Array`; and you explicitly state that its length should be greater 0 (same interpretation as just `leaf.obs.length`, but just more explicit and easier to read.

On further notice: This function should be refactored anyway. It uses `var` (bad, prefer `let`, but always try to use `const` if applicable), does not introduce any types (we have a lot of implicit `Array<any>`, and even a strange dictionary (`parents`) etc.). Furthermore nothing in this file is typed for some reason... It should definitely be improved to ensure proper usage.

Reply:
Also make `activateTests` a const, same with `activeLeaves`, `lowestParents` etc.

Reply:
Sure @FlorianRappl, I can go through for a revamp, but since I'm taken into some other work currently, @mks-d , please confirm the urgency on this, can this wait till the next week's end?

Thanks!

Reply:
Sure @FlorianRappl, I can go through for a revamp, but since I'm taken into some other work currently, @mks-d , please confirm the urgency on this, can this wait till the next week's end?

Thanks!

Reply:
There is no need to spend much time on this; just that you change the variable declarations for now. Adding a few types would also be beneficial and shouldn't take too long.

In any case - it's something we should do sooner or later.

Reply:
Thanks @FlorianRappl !